src/board.py

```diff
@@ -28,27 +35,15 @@ def __initialize_empty_board(self):
                     self.graph.add_edge(
                         (x, y), (x, y+1), direction=Board.SOUTH)
 
-    def display(self):
-        print('# '*(self.WIDTH+2))
-        for j in range(self.HEIGHT):
-            line = '# '
-            for i in range(self.WIDTH):
-                if (self.graph.nodes[(i, j)].get('puck')):
-                    line += 'o '
-                else:
-                    line += '  '
-            line += '#'
-            print(line)
-        print('# '*(self.WIDTH+2))
-
     def tilt(self, direction):
-        pucks = [node for node, attributes in self.graph.nodes(data=True) if attributes.get('puck')]
+        pucks = [node for node, attributes in self.graph.nodes(
+            data=True) if attributes.get('puck')]
         for puck in pucks:
             self.__move_puck_to(puck, direction)
 
     def __get_next_free_node(self, node, direction):
         next_node = [next_node for start_node, next_node, edge_attrs in self.graph.out_edges(
-            node, data=True) if edge_attrs.get('direction') == direction]
+            node, data=True) if (edge_attrs.get('direction') == direction and not self.graph.nodes[next_node].get('obstacle'))]
```

Comment:
you should remove the edges to obstacles instead

Reply:
done

Reply:
should be removed

Reply:
This time it's really gone